### PR TITLE
CI: add Scripting docs check

### PR DESCRIPTION
**Pull request by IamPete1**
This adds a check that scripting docs are upto date. It re-generates the docs using the generator and then spots the difference between the current docs and the generated ones. It ignores argument names and comments.

This should currently fail because there is a bunch of stuff missing. It will also fail because there is no definition for the expected number of arguments for manual bindings, this will also need to be added. 

Could expand further in the future to also check fields, argument/return types.

### CODE REVIEW

**Comment by IamPete1** [comment]
As expected:
```
Missing definition of:
function uint32_t() end
	Global: uint32_t
	Num Args: 0
Missing definition of:
function EFI_State_ud:pt_compensation(value) end
	Class: EFI_State_ud
	Function: pt_compensation
	Num Args: 1
Missing definition of:
function EFI_State_ud:throttle_out(value) end
	Class: EFI_State_ud
	Function: throttle_out
	Num Args: 1
Missing definition of:
function EFI_State_ud:ignition_voltage(value) end
	Class: EFI_State_ud
	Function: ignition_voltage
	Num Args: 1
Missing definition of:
function CANFrame_ud:id_signed() end
	Class: CANFrame_ud
	Function: id_signed
	Num Args: 0
Missing definition of:
function Parameter() end
	Global: Parameter
	Num Args: 0
Missing definition of:
function AP_EFI_Backend_ud:handle_scripting(param1) end
	Class: AP_EFI_Backend_ud
	Function: handle_scripting
	Num Args: 1
Missing definition of:
function efi:get_backend(param1) end
	Class: efi
	Function: get_backend
	Num Args: 1
Missing definition of:
function mount:get_attitude_euler(param1) end
	Class: mount
	Function: get_attitude_euler
	Num Args: 1
Missing definition of:
function vehicle:has_ekf_failsafed() end
	Class: vehicle
	Function: has_ekf_failsafed
	Num Args: 0
Missing definition of:
function vehicle:nav_script_time() end
	Class: vehicle
	Function: nav_script_time
	Num Args: 0
Missing definition of:
function ahrs:get_quaternion() end
	Class: ahrs
	Function: get_quaternion
	Num Args: 0
Missing definition of:
function ahrs:get_posvelyaw_source_set() end
	Class: ahrs
	Function: get_posvelyaw_source_set
	Num Args: 0
Missing definition of:
function ahrs:get_relative_position_D_home() end
	Class: ahrs
	Function: get_relative_position_D_home
	Num Args: 0
Missing definition of:
function ahrs:get_position() end
	Class: ahrs
	Function: get_position
	Num Args: 0
Unexpected definition of:
function uint32_t(value) end
	Global: uint32_t
	Num Args: 1
Unexpected definition of:
function uint32_t_ud:tofloat() end
	Class: uint32_t_ud
	Function: tofloat
	Num Args: 0
Unexpected definition of:
function uint32_t_ud:toint() end
	Class: uint32_t_ud
	Function: toint
	Num Args: 0
Unexpected definition of:
function millis() end
	Global: millis
	Num Args: 0
Unexpected definition of:
function micros() end
	Global: micros
	Num Args: 0
Unexpected definition of:
function mission_receive() end
	Global: mission_receive
	Num Args: 0
Unexpected definition of:
function logger:write(name, labels, format, units, multipliers, data1, ...) end
	Class: logger
	Function: write
	Num Args: [7](https://github.com/ArduPilot/ardupilot/actions/runs/3251170917/jobs/5335809377#step:6:8)
Unexpected definition of:
function i2c:get_device(bus, address, clock, smbus) end
	Class: i2c
	Function: get_device
	Num Args: 4
Unexpected definition of:
function efi:handle_scripting(efi_state) end
	Class: efi
	Function: handle_scripting
	Num Args: 1
Unexpected definition of:
function CAN:get_device(buffer_len) end
	Class: CAN
	Function: get_device
	Num Args: 1
Unexpected definition of:
function Parameter(name) end
	Global: Parameter
	Num Args: 1
Unexpected definition of:
function AP_HAL__I2CDevice_ud:read_registers(register_num, read_length) end
	Class: AP_HAL__I2CDevice_ud
	Function: read_registers
	Num Args: 2
Unexpected definition of:
function mount:get_attitude_euler(instance, roll_deg, pitch_deg, yaw_bf_deg) end
	Class: mount
	Function: get_attitude_euler
	Num Args: 4
Traceback (most recent call last):
  File "./libraries/AP_Scripting/tests/docs_check.py", line 12[8](https://github.com/ArduPilot/ardupilot/actions/runs/3251170917/jobs/5335809377#step:6:9), in <module>
    compare(args[0], args[1])
  File "./libraries/AP_Scripting/tests/docs_check.py", line [11](https://github.com/ArduPilot/ardupilot/actions/runs/3251170917/jobs/5335809377#step:6:12)4, in compare
    raise Exception("Docs do not match")
Exception: Docs do not match
```
Ran in 1 minute exactly.

**Comment by IamPete1** [comment]
Should now pass, added missing docs items and generation for manual bindings. Requires a number of arguments be given in the binding description to allow arg number check. Only thing that is not in the docs is operators, and i'm not sure how we would do those.

File: libraries/AP_Scripting/tests/docs_check.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
+'''
+Reads two lua docs files and checks for differences
+
+python ./libraries/AP_Scripting/tests/docs_check.py "./libraries/AP_Scripting/docs/docs expected.lua" "./libraries/AP_Scripting/docs/docs.lua"
+'''
```
Comment (peterbarker) [inline libraries/AP_Scripting/tests/docs_check.py:5]:
```suggestion

AP_FLAKE8_CLEAN
'''
```

File: libraries/AP_Scripting/tests/docs_check.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        print("Docs check passed")
+
+if __name__ == '__main__':
+
+    parser = optparse.OptionParser(__file__)
+
+    opts, args = parser.parse_args()
+
+    if len(args) != 2:
+        print("Usage: %s \"expected docs path\" \"current docs path\"" % parser.usage)
```
Comment (peterbarker) [inline libraries/AP_Scripting/tests/docs_check.py:125]:
```suggestion
        print('Usage: %s "expected docs path" "current docs path"' % parser.usage)
```

Comment (IamPete1) [reply libraries/AP_Scripting/tests/docs_check.py]: done

File: libraries/AP_Scripting/tests/docs_check.py (hunk 1 above)
Comment (IamPete1) [reply libraries/AP_Scripting/tests/docs_check.py:7]: I bet its not, I guess we'll find out....

**Comment by IamPete1** [comment]
Updated the action to only run for a change to scripting bindings or python checker.

**Review by peterbarker — APPROVED** [review]

**Review by tridge — APPROVED** [review]

**Pull request merged**